tests/framework/framework.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+var (
+	TestFramework       testFramework = noFrameworkSelected{}
+	RunE2eTests         testFramework = e2eFramework{}
+	RunIntegrationTests testFramework = integrationFramework{}
```

Comment:
The names `E2eFramework` and  `IntegrationFramework` make more sense than `RunE2eTests` and `RunIntegrationTests`

Reply:
Still working out the naming, I decided on this name because it's more readable from user perspective. As a user name `testFramework = framework.E2eFramework` doesn't really explain what the change does. On the other handl `testFramework = framework.RunE2eTests` explains that it will configure tests to run E2e scenario.

Reply:
I think going with `TestRunner` would be nicer. So `E2eTestRunner` and `IntegrationTestRunner`